### PR TITLE
Fix broken links for User Guide

### LINKED ISSUE #393: Fix Dead Links in MDAnalysis User Guide Documentation
I checked [MDAnalysis User Guide - Stable Index](https://userguide.mdanalysis.org/stable/index.html) and found several links that are either broken, lead to 404 errors, or are inaccessible. Below is a table summarizing the problematic links, including the pages where they are found and the anchor text associated with them.

| **#** | **Error**       | **Anchor Text**         | **Linked From**                                               | **Broken Link**                                                                                  |
|-------|-----------------|-------------------------|---------------------------------------------------------------|--------------------------------------------------------------------------------------------------|
| 1     | 404 Not Found   | HOLE2 MDAKit            | [Releases](https://userguide.mdanalysis.org/stable/releases.html)  | [https://www.mdanalysis.org/hole2-mdakit/](https://www.mdanalysis.org/hole2-mdakit/)            |
| 2     | 404 Not Found   | @g2707                  | [Releases](https://userguide.mdanalysis.org/stable/releases.html)  | [https://github.com/g2707](https://github.com/g2707)                                            |
| 3     | 404 Not Found   | align_structure         | [Examples - README](https://userguide.mdanalysis.org/stable/examples/analysis/README.html) | [https://userguide.mdanalysis.org/examples/analysis/alignment_and_rms/aligning_structure_to_another.ipynb](https://userguide.mdanalysis.org/examples/analysis/alignment_and_rms/aligning_structure_to_another.ipynb) |
| 4     | 404 Not Found   | align_trajectory_first  | [Examples - README](https://userguide.mdanalysis.org/stable/examples/analysis/README.html) | [https://userguide.mdanalysis.org/examples/analysis/alignment_and_rms/aligning_trajectory_to_first_frame.ipynb](https://userguide.mdanalysis.org/examples/analysis/alignment_and_rms/aligning_trajectory_to_first_frame.ipynb) |
| 5     | 404 Not Found   | align_trajectory        | [Examples - README](https://userguide.mdanalysis.org/stable/examples/analysis/README.html) | [https://userguide.mdanalysis.org/examples/analysis/alignment_and_rms/aligning_trajectory.ipynb](https://userguide.mdanalysis.org/examples/analysis/alignment_and_rms/aligning_trajectory.ipynb) |
| 6     | 404 Not Found   | rmsf                    | [Examples - README](https://userguide.mdanalysis.org/stable/examples/analysis/README.html) | [https://userguide.mdanalysis.org/examples/analysis/alignment_and_rms/rmsf.ipynb](https://userguide.mdanalysis.org/examples/analysis/alignment_and_rms/rmsf.ipynb) |
| 7     | 404 Not Found   | Pydagogue               | [Contributing](https://userguide.mdanalysis.org/stable/contributing.html) | [https://matthew-brett.github.com/pydagogue/](https://matthew-brett.github.com/pydagogue/)       |
| 8     | 404 Not Found   | DataFrame               | [RMSD Analysis](https://userguide.mdanalysis.org/stable/examples/analysis/alignment_and_rms/rmsd.html) | [https://pandas.pydata.org/pandas-docs/stable/getting_started/dsintro.html](https://pandas.pydata.org/pandas-docs/stable/getting_started/dsintro.html) |
| 9     | -1 Not Found    | MMTF                    | [MMTF Format](https://userguide.mdanalysis.org/stable/formats/reference/mmtf.html) | [https://mmtf.rcsb.org/](https://mmtf.rcsb.org/)                                                |
| 10    | -1 Not Found    | HOOMD-blue              | [GSD Format](https://userguide.mdanalysis.org/stable/formats/reference/gsd.html) | [http://codeblue.umich.edu/hoomd-blue/index.html](http://codeblue.umich.edu/hoomd-blue/index.html) |
| 11    | -1 Not Found    | XML                     | [XML Format](https://userguide.mdanalysis.org/stable/formats/reference/xml.html) | [http://codeblue.umich.edu/hoomd-blue/doc/page_xml_file_format.html](http://codeblue.umich.edu/hoomd-blue/doc/page_xml_file_format.html) |
| 12    | -1 Not Found    | GAMESS-UK               | [GMS Format](https://userguide.mdanalysis.org/stable/formats/reference/gms.html) | [http://www.cfs.dl.ac.uk/](http://www.cfs.dl.ac.uk/)                                            |
| 13    | 404 Not Found   | MOL2                    | [MOL2 Format](https://userguide.mdanalysis.org/stable/formats/reference/mol2.html) | [http://chemyang.ccnu.edu.cn/ccb/server/AIMMS/mol2.pdf](http://chemyang.ccnu.edu.cn/ccb/server/AIMMS/mol2.pdf) |
| 14    | -1 Not Found    | FHI-aims                | [IN Format](https://userguide.mdanalysis.org/stable/formats/reference/in.html) | [https://aimsclub.fhi-berlin.mpg.de/](https://aimsclub.fhi-berlin.mpg.de/)                       |
| 15    | 404 Not Found   | PQR                     | [PQR Format](https://userguide.mdanalysis.org/stable/formats/reference/pqr.html) | [https://apbs-pdb2pqr.readthedocs.io/en/latest/formats/pqr.html](https://apbs-pdb2pqr.readthedocs.io/en/latest/formats/pqr.html) |
| 16    | 404 Not Found   | PDB2PQR                 | [PQR Format](https://userguide.mdanalysis.org/stable/formats/reference/pqr.html) | [https://apbs-pdb2pqr.readthedocs.io/en/latest/pdb2pqr/index.html](https://apbs-pdb2pqr.readthedocs.io/en/latest/pdb2pqr/index.html) |
| 17    | 404 Not Found   | APBS                    | [PQR Format](https://userguide.mdanalysis.org/stable/formats/reference/pqr.html) | [https://apbs-pdb2pqr.readthedocs.io/en/latest/apbs/index.html](https://apbs-pdb2pqr.readthedocs.io/en/latest/apbs/index.html) |
| 18    | -1 Not Found    | CRD                     | [CRD Format](https://userguide.mdanalysis.org/stable/formats/reference/crd.html) | [https://www.charmmtutorial.org/index.php/CHARMM:The_Basics](https://www.charmmtutorial.org/index.php/CHARMM:The_Basics) |
| 19    | 404 Not Found   | PARM                    | [TOP Format](https://userguide.mdanalysis.org/stable/formats/reference/top.html) | [https://ambermd.org/formats.html](https://ambermd.org/formats.html)                             |
| 20    | -1 Not Found    | IBIsCO                  | [TRZ Format](https://userguide.mdanalysis.org/stable/formats/reference/trz.html) | [http://www.theo.chemie.tu-darmstadt.de/ibisco/IBISCO.html](http://www.theo.chemie.tu-darmstadt.de/ibisco/IBISCO.html) |
| 21    | -1 Not Found    | YASP                    | [TRZ Format](https://userguide.mdanalysis.org/stable/formats/reference/trz.html) | [http://www.theo.chemie.tu-darmstadt.de/group/services/yaspdoc/yaspdoc.html](http://www.theo.chemie.tu-darmstadt.de/group/services/yaspdoc/yaspdoc.html) |
                    
I am ready to start working on fixing these issues, but I would appreciate some feedback or suggestions before proceeding. Also, if there are preferred sources to replace outdated links, please provide them.

### PR DESCRIPTION
**Pull request by mhmohona**
Fixed broken links for User Guide

Close: #393 

----
📚 Documentation preview 📚: https://mdanalysisuserguide--394.org.readthedocs.build/en/394/

### CODE REVIEW

File: doc/source/examples/analysis/alignment_and_rms/README.rst [hunk 1]
```diff
@@ -10,10 +10,10 @@ The :mod:`MDAnalysis.analysis.align` and :mod:`MDAnalysis.analysis.rms`
 modules contain the functions used for aligning structures,
 aligning trajectories, and calculating root mean squared quantities.
 
-Demonstrations of alignment are in `align_structure`_,
-`align_trajectory_first`_, and `align_trajectory`_. Another example of
+Demonstrations of alignment are in `align_structure <aligning_structure_to_another.ipynb>`__,
+`align_trajectory_first <aligning_trajectory_to_first_frame.ipynb>`__, and `align_trajectory <aligning_trajectory.ipynb>`__. Another example of
```
Comment (orbeckst) [inline doc/source/examples/analysis/alignment_and_rms/README.rst]: change `aligning_trajectory_to_first_frame.ipynb` to `aligning_trajectory_to_frame.ipynb` and double check that the link resolves

File: doc/source/examples/analysis/alignment_and_rms/README.rst [hunk 2]
```diff
@@ -33,10 +33,5 @@ RMSF values to make sense.
    /examples/analysis/alignment_and_rms/pairwise_rmsd
    /examples/analysis/alignment_and_rms/rmsf
 
-
-.. _align_structure: /examples/analysis/alignment_and_rms/aligning_structure_to_another.ipynb
```
Comment (orbeckst) [inline doc/source/examples/analysis/alignment_and_rms/README.rst:37]: Doing inline links is fine but did you try to just use something like
```reST
.. _align_structure: aligning_structure_to_another.ipynb
```

File: doc/source/releases.md [hunk 3]
```diff
@@ -218,26 +218,26 @@ See the [CHANGELOG](https://github.com/MDAnalysis/mdanalysis/blob/release-2.5.0/
 * [@jvermaas](https://github.com/jvermaas) made their first contribution in https://github.com/MDAnalysis/mdanalysis/pull/4031
 * [@SophiaRuan](https://github.com/SophiaRuan) made their first contribution in https://github.com/MDAnalysis/mdanalysis/pull/4072
 * [@marinegor](https://github.com/marinegor) made their first contribution in https://github.com/MDAnalysis/mdanalysis/pull/4085
-* [@g2707](https://github.com/g2707) made their first contribution in https://github.com/MDAnalysis/mdanalysis/pull/4089
-* [@DanielJamesEvans](https://github.com/DanielJamesEvans) made their first contribution in https://github.com/MDAnalysis/mdanalysis/pull/4109
-
+* [@shaivimalik](https://github.com/shaivimalik) made their first contribution in https://github.com/MDAnalysis/mdanalysis/pull/4089
```
Comment (orbeckst) [inline doc/source/releases.md]: That's some link-hunting for finding the changed GH name via the PR :-)

File: doc/source/contributing.rst [hunk 4]
```diff
@@ -75,7 +75,7 @@ Some great resources for learning Git:
 
     * the `GitHub help pages <http://help.github.com/>`_.
     * the `NumPy's documentation <http://docs.scipy.org/doc/numpy/dev/index.html>`_.
-    * Matthew Brett's `Pydagogue <http://matthew-brett.github.com/pydagogue/>`_.
+    * Matthew Brett's `The curious coder’s guide to git <https://matthew-brett.github.io/curious-git/>`_.
```
Comment (orbeckst) [inline doc/source/contributing.rst]: I think the replacement link should be https://matthew-brett.github.io/pydagogue/ ; the Curious Git guide is also linked from the pydagogue page.

**Review by orbeckst — CHANGES_REQUESTED** [review]
Thank you for your link updates. Sorry for the long delay in giving you feedback.

Please see my comments inline. Thank you!

**Comment by orbeckst** [comment]
Please fix the formatting that makes the pre-commit check fail https://github.com/MDAnalysis/UserGuide/actions/runs/12404131142/job/34628800686?pr=394

**Comment by orbeckst** [comment]
You can install pre-commit yourself and let it fix these things, see https://userguide.mdanalysis.org/stable/contributing_docs.html#using-pre-commit-hooks

**Comment by mhmohona** [comment]
Hello @orbeckst, thank you for your help. I have fixed the precommit issue. Can you please have a look at it again?

File: doc/source/contributing.rst [hunk 5]
```diff
@@ -75,7 +75,7 @@ Some great resources for learning Git:
 
     * the `GitHub help pages <http://help.github.com/>`_.
     * the `NumPy's documentation <http://docs.scipy.org/doc/numpy/dev/index.html>`_.
-    * Matthew Brett's `Pydagogue <http://matthew-brett.github.com/pydagogue/>`_.
+    * Matthew Brett's `The curious coder’s guide to git <https://matthew-brett.github.io/pydagogue/>`_.
```
Comment (orbeckst) [inline doc/source/contributing.rst]: wrong name, it's still Pydagogue

File: doc/source/examples/analysis/alignment_and_rms/README.rst [hunk 6]
```diff
@@ -10,10 +10,10 @@ The :mod:`MDAnalysis.analysis.align` and :mod:`MDAnalysis.analysis.rms`
 modules contain the functions used for aligning structures,
 aligning trajectories, and calculating root mean squared quantities.
 
-Demonstrations of alignment are in `align_structure`_,
-`align_trajectory_first`_, and `align_trajectory`_. Another example of
+Demonstrations of alignment are in `align_structure <aligning_structure_to_another.ipynb>`__,
+`align_trajectory_first <aligning_trajectory_to_frame.ipynb>`__, and `align_trajectory <aligning_trajectory.ipynb>`__. Another example of
```
Comment (orbeckst) [inline doc/source/examples/analysis/alignment_and_rms/README.rst:14]: check links on https://mdanalysisuserguide--394.org.readthedocs.build/en/394/examples/analysis/alignment_and_rms/README.html ... not all of them work

**Review by orbeckst — CHANGES_REQUESTED** [review]

File: doc/source/examples/analysis/alignment_and_rms/README.rst (hunk 6 above)
Comment (mhmohona) [reply doc/source/examples/analysis/alignment_and_rms/README.rst:14]: I just checked all the links here - https://mdanalysisuserguide--394.org.readthedocs.build/en/394/examples/analysis/alignment_and_rms/README.html and found they are working. Can you please check again?

Comment (orbeckst) [reply doc/source/examples/analysis/alignment_and_rms/README.rst:14]: [align_trajectory_first](https://mdanalysisuserguide--394.org.readthedocs.build/en/394/examples/analysis/alignment_and_rms/aligning_trajectory_to_first_frame.ipynb) is 404

Comment (mhmohona) [reply doc/source/examples/analysis/alignment_and_rms/README.rst:14]: > [align_trajectory_first](https://mdanalysisuserguide--394.org.readthedocs.build/en/394/examples/analysis/alignment_and_rms/aligning_trajectory_to_first_frame.ipynb) is 404

So I am getting this link after clicking on `aline_trajectory_first` which is working - https://mdanalysisuserguide--394.org.readthedocs.build/en/394/examples/analysis/alignment_and_rms/aligning_trajectory_to_frame.html. 
I also tried different browser, and its working here.

Comment (orbeckst) [reply doc/source/examples/analysis/alignment_and_rms/README.rst:14]: The dead link is in

> Demonstrations of alignment are in [align_structure](https://mdanalysisuserguide--394.org.readthedocs.build/en/394/examples/analysis/alignment_and_rms/aligning_structure_to_another.html), [align_trajectory_first](https://mdanalysisuserguide--394.org.readthedocs.build/en/394/examples/analysis/alignment_and_rms/aligning_trajectory_to_first_frame.ipynb), and [align_trajectory](https://mdanalysisuserguide--394.org.readthedocs.build/en/394/examples/analysis/alignment_and_rms/aligning_trajectory.html). Another example of generating an average structure from an alignment is demonstrated in [rmsf](https://mdanalysisuserguide--394.org.readthedocs.build/en/394/examples/analysis/alignment_and_rms/rmsf.html). Typically, trajectories need to be aligned for RMSD and RMSF values to make sense.

Comment (orbeckst) [reply doc/source/examples/analysis/alignment_and_rms/README.rst:14]: ... but after hard-refreshing I now get the working link!

Ok, so this is good now. Thank you for telling me your side.

**Review by orbeckst — APPROVED** [review]
All looking good, thank you for promptly addressing my requested changes.

Thanks a lot for your contribution (and your patience!!).

**Pull request merged**